pyk/src/pyk/prelude/ml.py

```diff
@@ -20,12 +22,24 @@
 }
 
 
-def is_top(term: KInner) -> bool:
-    return isinstance(term, KApply) and term.label.name == '#Top'
-
-
-def is_bottom(term: KInner) -> bool:
-    return isinstance(term, KApply) and term.label.name == '#Bottom'
+def is_top(term: KInner, weak: bool = False) -> bool:
+    if weak:
+        flat = flatten_label('#And', term)
+        if len(flat) == 1:
+            return is_top(single(flat))
+        return all(is_top(term, weak=True) for term in flat)
+    else:
+        return isinstance(term, KApply) and term.label.name == '#Top'
```

Comment:
Consider extracting `_is_top` for the simple condition (as in an earlier version).